snap/snapcraft.yaml

```diff
@@ -277,26 +271,47 @@ parts:
       - libattr1-dev
       - libfdt-dev
       - curl
+      - libcapstone-dev
     build-snaps:
       - yq
     override-build: |
-      chmod +x ${SNAPCRAFT_STAGE}/qemu/scripts/configure-hypervisor.sh
-      # static build
-      echo "$(${SNAPCRAFT_STAGE}/qemu/scripts/configure-hypervisor.sh -s qemu) \
-        --disable-rbd
-        --prefix=/snap/${SNAPCRAFT_PROJECT_NAME}/current/usr \
-        --datadir=/snap/${SNAPCRAFT_PROJECT_NAME}/current/usr/share \
-        --libexecdir=/snap/${SNAPCRAFT_PROJECT_NAME}/current/usr/libexec/qemu" \
-        | xargs ./configure
+      # set GOPATH
+      export GOPATH=$(realpath go)
+      # go was built in parts/go/build
+      export GOROOT=$(realpath ../../go/build)
+      export PATH="${GOROOT}/bin:${PATH}"
+
+      versions_url=https://raw.githubusercontent.com/kata-containers/runtime/${SNAPCRAFT_PROJECT_VERSION}/versions.yaml
+      # arch-specific definition
+      case "$(arch)" in
+        "aarch64")
+          branch="$(curl -sSL ${versions_url} | yq r - assets.hypervisor.qemu.architecture.aarch64.branch)"
+          url="$(curl -sSL ${versions_url} | yq r - assets.hypervisor.qemu.url)"
+          commit="$(curl -sSL ${versions_url} | yq r - assets.hypervisor.qemu.architecture.aarch64.commit)"
+          pkg_gopath="${GOPATH}/src/github.com/qemu/qemu"
+          patch_dir="${SNAPCRAFT_STAGE}/qemu-aarch64/patches/"
+        ;;
 
-      git clone https://github.com/qemu/keycodemapdb ui/keycodemapdb/
-      cd ui/keycodemapdb; git checkout 10739aa; cd ../..
+        *)
+          branch="$(curl -sSL ${versions_url} | yq r - assets.hypervisor.qemu-lite.branch)"
+          url="$(curl -sSL ${versions_url} | yq r - assets.hypervisor.qemu-lite.url)"
+          commit="$(curl -sSL ${versions_url} | yq r - assets.hypervisor.qemu-lite.commit)"
+          pkg_gopath="${GOPATH}/src/github.com/kata-containers/qemu"
+          patch_dir="${SNAPCRAFT_STAGE}/qemu-lite/patches/"
+        ;;
+      esac
+
+      mkdir -p ${pkg_gopath}
+      # download source
+      git clone --branch ${branch} --single-branch ${url} ${pkg_gopath}
```

Comment:
you can clone and build qemu in the current directory
`git clone --branch ${branch} --single-branch ${url} .`